lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py

```diff
@@ -294,3 +283,124 @@ def test_job_non_existing(local_client: TestClient, dependency_overrides_service
     response = local_client.get(f"/jobs/{non_existing_id}")
     assert response.status_code == 404
     assert response.json()["detail"] == f"Job with ID {non_existing_id} not found"
+
+
+def test_create_exp_workflow_check_results(
+    local_client: TestClient, dialog_dataset, dependency_overrides_services
+):
+    """Here's how it will work
+    * Create a dataset
+    * Create an experiment, get back an ID and validate it
+    * Create a workflow for that experiment
+    * Poll the status of the workflow until it's done
+    * Get the results of the workflow
+    """
+    # Make sure backend is healthy
+    response = local_client.get("/health")
+    assert response.status_code == 200
+
+    # Upload a dataset
+    create_response = local_client.post(
+        "/datasets/",
+        data={},
+        files={"dataset": dialog_dataset, "format": (None, DatasetFormat.JOB.value)},
+    )
+    assert create_response.status_code == 201
+
+    dataset = DatasetResponse.model_validate(create_response.json())
+
+    experiment = local_client.post(
+        "/experiments/new/",
+        headers=POST_HEADER,
+        json={
+            "name": "test_create_exp_workflow_check_results",
+            "description": "Test for an experiment with associated workflows",
+        },
+    )
+    assert experiment.status_code == 201
+    experiment_id = experiment.json()["id"]
+
+    # run a workflow for that experiment
+    workflow_1 = WorkflowResponse.model_validate(
+        local_client.post(
+            "/workflows/",
+            headers=POST_HEADER,
+            json={
+                "name": "Workflow_1",
+                "description": "Test workflow for inf and eval",
+                "model": TEST_CAUSAL_MODEL,
+                "dataset": str(dataset.id),
+                "experiment_id": experiment_id,
+                "max_samples": 1,
+            },
+        ).json()
+    )
+
+    # Wait till the workflow is done
+    workflow_status = workflow_1.status
```

Comment:
Put a max number of polls, otherwise we can be spending millions on CI :D

Reply:
💵 💸 
I'll update, good call